src/ui/public/vis/update_status.js

```diff
@@ -0,0 +1,47 @@
+const oldValues = [];
+
+// adapted from https://github.com/isaacs/json-stringify-safe/blob/02cfafd45f06d076ac4bf0dd28be6738a07a72f9/stringify.js
+function serializer() {
+  const stack = [];
+  const keys = [];
+
+  const cycleReplacer = function (key, value) {
+    if (stack[0] === value) return '[Circular ~]';
+    return `[Circular ~.${keys.slice(0, stack.indexOf(value)).join('.')}]`;
+  };
+
+  return function (key, value) {
+    if (stack.length > 0) {
+      const thisPos = stack.indexOf(this);
+      ~thisPos ? stack.splice(thisPos + 1) : stack.push(this);
+      ~thisPos ? keys.splice(thisPos, Infinity, key) : keys.push(key);
+      if (~stack.indexOf(value)) value = cycleReplacer.call(this, key, value);
+    }
+    else stack.push(value);
+
+    return value;
+  };
+}
+
+const hasChanged = (name, value) => {
+  const currentValue = JSON.stringify(value, serializer());
+  if (currentValue !== oldValues[name]) {
+    oldValues[name] = currentValue;
+    return false;
+  }
+  return true;
+};
+
+const getUpdateStatus = ($scope) => {
```

Comment:
I like this idea, and it seems to work well.

I think there is a bug though with adding a map-visualization to the dashboard.

When the dashboard loads the first time, the map is blank. This is because the `data`, `uiState` and `resize` flag are both false.
![image](https://user-images.githubusercontent.com/1833023/27894660-895082e6-61db-11e7-9ddf-e2947655221b.png)